src/lib.rs

```diff
@@ -575,6 +576,51 @@ mod tests {
         assert_eq!(bytes.as_slice_of::<u64>(), Ok(slice.as_ref()));
     }
 
+    #[test]
+    fn usize() {
+        let slice: [usize; 2] = [0, 1];
+        let bytes = slice.as_byte_slice();
+
+        if cfg!(target_endian = "big") {
+            if cfg!(target_pointer_width = "16") {
```

Comment:
There's 16 bit pointer support in Rust now? Nice :) DOS or for some microcontroller or ...?

Reply:
I determined all the supported "usize" values reading https://doc.rust-lang.org/src/core/num/mod.rs.html#4566

I'm unsure which 16 bits platform are supported but at the moment, with "std" in the mix, I feel that the generated binary is too large to support a 16 bits platform properly. If I was to do DOS development, I would probably stick to C for the time being, even if I find Rust generally great!